pipeline-runner/R/gem2s-4-score_doublets.R

```diff
@@ -1,68 +1,65 @@
-#  - Compute doublet scores per sample
-#  - Save an txt file with doublet scores per sample [doublet-scores-sampleName]
-
-score_doublets <- function(input, pipeline_config) {
-  print(list.files(paste("/output", sep = "/"), all.files = TRUE, full.names = TRUE, recursive = TRUE))
-  message("reloading old matrices...")
-
-  scdata_list <- readRDS("/output/pre-doublet-scdata_list.rds")
-
-  message("Loading configuration...")
-  config <- RJSONIO::fromJSON("/input/meta.json")
-
-  print_config(4, "Score Doublets", input, pipeline_config, config)
-
-  # Check which samples have been selected. Otherwiser we are going to use all of them.
-  if (length(config$samples) > 0) {
-    samples <- config$samples
-  } else {
-    samples <- names(scdata_list)
-  }
-
-  scdata_list <- scdata_list[samples]
-
-  message("calculating probability of barcodes being doublets...")
-  for (sample_name in names(scdata_list)) {
-    compute_doublet_scores(scdata_list[[sample_name]], sample_name)
+#' Compute doublet score for count matrices
+#'
+#' @inheritParams download_cellranger
+#' @param prev_out 'output' slot from call to \code{run_emptydrops}
+#'
+#' @return \code{prev_out} with added slot 'doublet_scores' containing result of call to
+#'   \code{\link{compute_doublet_scores}} for each sample.
+#'
+#' @export
+#'
+score_doublets <- function(input, pipeline_config, prev_out) {
+  message("Calculating probability of droplets being doublets...")
+
+  edrops_list <- prev_out$edrops
+  counts_list <- prev_out$counts_list
```

Comment:
Are you validating and checking anywhere that these actually exist in `prev_out`? If step 3 was disabled or didn't filter anything, wouldn't these lines of code here break?

Reply:
If edrops isn't there it will be fine (`edrops_list` will be `NULL`) (in fact we expect it to not be there if it is a filtered dataset).  If `counts_list` isn't there it will fail as it should because it means we either don't have any count matrices or we didn't pass them on and there is no way for GEM2S to complete. I'll add a check for it.